clang/include/clang/Basic/AttrDocs.td

```diff
@@ -1993,6 +1993,22 @@ device kernel, the attribute is ignored and it is not propagated to a kernel.
   }];
 }
 
+def SYCLIntelMaxGlobalWorkDimAttrDocs : Documentation {
+  let Category = DocCatFunction;
+  let Heading = "max_global_work_dim (IntelFPGA)";
+  let Content = [{
+Applies to a device function/lambda function or function call operator (of a
+function object). Indicates the largest valid global work dimension that will be
+accepted when running the kernel on a device. Valid values are integers in a
+range of [0, 3]. A kernel with max_global_work_dim(0) must be invoked with a
+'single_task' and if ``intelfpga::max_work_group_size`` or
```

Comment:
> A kernel with max_global_work_dim(0) must be invoked with a
> 'single_task'

Is there a way to enforce/test this? I lack knowledge if this is already implicit.

Reply:
Not on the SYCL layer. This information is for users mostly.

Reply:
This is why I think attributes are bad™ and should not be used in SYCL as I have explained already in some various presentations at Khronos in the past.